extensions/amp-form/0.1/amp-form.js

```diff
@@ -72,13 +79,27 @@ export class AmpForm {
     }
     if (this.xhrAction_) {
       e.preventDefault();
+      this.setState_(FormState_.SUBMITTING);
       this.xhr_.fetchJson(this.xhrAction_, {
         body: new FormData(this.form_),
         method: this.method_,
         credentials: 'include',
         requireAmpResponseSourceOrigin: true,
-      });
+      }).then(() => this.setState_(FormState_.SUBMIT_SUCCESS))
+          .catch(() => this.setState_(FormState_.SUBMIT_ERROR));
+    }
+  }
+
+  /**
+   * Adds proper classes for the state passed.
+   * @param {string} state
+   * @private
+   */
+  setState_(state) {
+    for (const key in FormState_) {
```

Comment:
@cramforce's mentioned before that Closure screws up with weird introspection. Can you confirm this compiles correctly?

If not, we can get around it by storing the current state.